core/query/query.mli

```diff
@@ -21,8 +21,9 @@ sig
       | Erase     of t * StringSet.t
       | Variant   of string * t
       | XML       of Value.xmlitem
-      | Apply     of string * t list
-      | Closure   of (Ir.var list * Ir.computation) * env
+      | Apply     of t * t list
+      | Closure   of (Var.var list * Ir.computation) * env
```

Comment:
This seems to get us epsilon closer to translating `Ir` expressions to queries with no dangling `Ir` references.  If we're handing `Case` constructs, how much harder would it be to handle function bodies/closures the same way?  It seems like the main issue is making sure that the translation does the right thing when translating under a binder (when encountering variables that aren't bound to values in `env`), which is already something that has to work right for `For` and `Case`.

Not something to address here necessarily, just a thought.